File: trunk/ShadowFrameworkLite20/src/shadow/pipeline/expression/SFExpressionVariable.java
package shadow.pipeline.expression;

import shadow.math.SFValuenf;
import shadow.pipeline.parameters.SFParameteri;
import shadow.system.SFException;

public class SFExpressionVariable extends SFExpressionElement {

	public SFExpressionVariable(String element, short type) {
		super(element);
		setType(type);
	}
	
	@Override
	public SFExpressionElement cloneAsIndexed(SFParameteri[] toBeIndexed) {
		for (int i = 0; i < toBeIndexed.length; i++) {
			if(toBeIndexed[i].getName().equalsIgnoreCase(getElement())){
				return new SFExpressionIndex(i);
			}
		}
		return this;
	}

	@Override
	public void addSubExpression(SFExpressionElement element) throws ArrayIndexOutOfBoundsException {
		// Nothing to do
	}

	@Override
	public void evaluateType() throws SFExpressionException {
		// Nothing to do..
	}

	@Override
	public SFValuenf evaluate(SFExpressionValuesList values) {

		try {
			SFValuenf value = values.generateValue();
			double d = new Double(getElement());
			for (int i = 0; i < value.get().length; i++) {
				value.get()[i] = (float)d;
			}
			return value;
		} catch (Exception e) {
			e.printStackTrace();
			throw new SFException("Cannot evaluate unindexed value "+getElement());
		}
	}

}

File: trunk/ShadowFrameworkJS/src/shadow/pipeline/expression/SFExpressionVariable.js

function SFExpressionVariable(element,type){
	this.list=new Array();
	this.element=element;
	this.setType(type);
}

inherit(SFExpressionVariable,SFExpressionElement);


SFExpressionVariable.prototype["cloneAsIndexed"]=function(toBeIndexed){
	for (var i = 0; i < toBeIndexed.length; i++) {
		if(toBeIndexed[i].getName()===(this.getElement())){
			return new SFExpressionIndex(i);
		}
	}
	return this;
};
		
SFExpressionVariable.prototype["addSubExpression"]=function(element){
			//Nothing to do;
		};
	
SFExpressionVariable.prototype["evaluateType"]=function(){
			//Nothing to do;
		};
		
SFExpressionVariable.prototype["evaluate"]=function(values){

			var value = values.generateValue();
			var d = getElement();
			for (var i = 0; i < value.get().length; i++) {
				value.get()[i] = d;
			}
			return value;
			
		};